lib/middleware.ts

```diff
@@ -27,6 +27,16 @@ interface MiddlewareOptions {
   liveReloadPath?: string;
 }
 
+function findClosestIndexFileForPath(outputPath: string, prefix: string): string | undefined {
```

Comment:
@stefanpenner we could skip index search if path contain `.` to prevent traversal security issues.

I seen cases where multiple static apps composed into one using nesting (and have to deal with it):

```js
root_app
   /child-app
   /some-side-app
   /help-app
```

Reply:
Ya it’s not a bad feature at all.

Reply:
@stefanpenner added guard for `.` in path <https://github.com/broccolijs/broccoli/pull/491/commits/99b5a323fc82560c6d65548ca893afb3b8ff764e#diff-92fc8ac710621cc64e5b3683c8c56bf5aec7a209c2968a1818bab04b11551811R87> and improved tests